File: src/test/scala/photos/brooklyn/practice/codingexercises/scala/leetcode/BuddyStringsTest.scala
package photos.brooklyn.practice.codingexercises.scala.leetcode

import org.scalatest.{FlatSpec, Matchers}

import BuddyStrings._

class BuddyStringsTest extends FlatSpec with Matchers {

  behavior of "BuddyStringsTest"

  it should "buddyStrings reversed length 2" in {
    buddyStrings("ab", "ba") shouldEqual true
  }

  it should "buddyStrings same length 2" in {
    buddyStrings("ab", "ab") shouldEqual false
  }

  it should "buddyStrings same with repeats" in {
    buddyStrings("aa", "aa") shouldEqual true
  }

  it should "buddyStrings diff last 2" in {
    buddyStrings("aaaaaaabc", "aaaaaaacb") shouldEqual true
  }

  it should "buddyStrings diff length" in {
    buddyStrings("", "aa") shouldEqual false
  }

}

File: test/java/photos/brooklyn/practice/codingexercises/java/leetcode/BuddyStringsTest.java
package photos.brooklyn.practice.codingexercises.java.leetcode;

import org.junit.Test;

import static org.junit.Assert.*;

public class BuddyStringsTest {

    @Test
    public void buddyStringsReversed() {
        assertTrue(new BuddyStrings().buddyStrings("ab", "ba"));
    }
    @Test
    public void buddyStringsSame() {
        assertFalse(new BuddyStrings().buddyStrings("ab", "ab"));
    }
    @Test
    public void buddyStringsSameRepeat() {
        assertTrue(new BuddyStrings().buddyStrings("aa", "aa"));
    }
    @Test
    public void buddyStringsDiffEnd() {
        assertTrue(new BuddyStrings().buddyStrings("aaaaaaabc", "aaaaaaacb"));
    }
    @Test
    public void buddyStringsDiffLength() {
        assertFalse(new BuddyStrings().buddyStrings("", "aa"));
    }
}